### PR TITLE
add #[panic_handler]; deprecate #[panic_implementation]

### PR DESCRIPTION
**Pull request by japaric**
r? @SimonSapin 
cc #44489

### CODE REVIEW

**Comment by SimonSapin** [comment]
Oh nice, I didn’t realize we already had a mechanism for deprecated attributes.

Looks great, r=me with the two tidy errors fixed.

@bors delegate+

File: src/libsyntax/feature_gate.rs [hunk 1]
```diff
@@ -1102,11 +1103,18 @@ pub const BUILTIN_ATTRIBUTES: &'static [(&'static str, AttributeType, AttributeG
                                    "infer 'static lifetime requirements",
                                    cfg_fn!(infer_static_outlives_requirements))),
 
+    // RFC 2070 (deprecated attribute name)
+    ("panic_implementation",
+     Normal, Gated(Stability::Deprecated("https://github.com/rust-lang/rust/issues/44489#issuecomment-415140224"),
```
Comment (SimonSapin) [inline src/libsyntax/feature_gate.rs]: Just to double check: `Stability::Deprecated` implies unstable, right?

Comment (japaric) [reply src/libsyntax/feature_gate.rs]: You still need `#![feature(panic_implementation)]` to use the deprecated `#[panic_implementation]` attribute (see recently added ui / feature-gate test) so it remains a nightly-only feature.

**Comment by japaric** [comment]
I'll fix the Travis failure later today, until then

@bors r-

**Pull request merged**